core/src/main/scala/akka/persistence/r2dbc/query/scaladsl/R2dbcReadJournal.scala

```diff
@@ -110,19 +113,55 @@ final class R2dbcReadJournal(system: ExtendedActorSystem, config: Config, cfgPat
 
   private val filteredPayloadSerId = SerializationExtension(system).findSerializerFor(FilteredPayload).identifier
 
+  // key is tuple of entity type and slice
+  private val heartbeatPersistenceIds = new ConcurrentHashMap[(String, Int), String]()
+  private val heartbeatUuid = UUID.randomUUID().toString
+  log.debug("Using heartbeat UUID [{}]", heartbeatUuid)
+
+  private def heartbeatPersistenceId(entityType: String, slice: Int): String = {
+    val key = entityType -> slice
+    heartbeatPersistenceIds.get(key) match {
+      case null =>
+        // no need to block other threads, it's just a cache
+        val pid = generateHeartbeatPersistenceId(entityType, slice)
+        heartbeatPersistenceIds.put(key, pid)
+        pid
+      case pid => pid
+    }
+  }
+
+  @tailrec private def generateHeartbeatPersistenceId(entityType: String, slice: Int, n: Int = 1): String = {
+    if (n < 1000000) {
+      // including a uuid to make sure it is not the same as any persistence id of the application
+      val pid = PersistenceId.concat(entityType, s"_hb-$heartbeatUuid-$n")
```

Comment:
I added a uuid in the pid. Shouldn't be high frequency of these. Now it's one per idle tick (per 3 seconds), but we could emit less often. Could be nice to be able to track correlation via an uuid as well. Not a big deal, could change to `s"_hb-akka-$n"`

Reply:
Seems ok. I guess we could also make it more unique with its format and mark as akka internal. But can be revisited at any time.